src/parameterizations/vertical/MOM_set_viscosity.F90

```diff
@@ -1202,6 +1086,707 @@ subroutine set_viscous_BBL(u, v, h, tv, visc, G, GV, US, CS, pbv)
 
 end subroutine set_viscous_BBL
 
+!> Determine the normalized open length of each interface, given the edge depths and normalized
+!! volumes below each interface.
+subroutine find_L_open_uniform_slope(vol_below, Dp, Dm, L, GV)
+  type(verticalGrid_type),     intent(in)  :: GV   !< The ocean's vertical grid structure.
+  real, dimension(SZK_(GV)+1), intent(in)  :: vol_below !< The volume below each interface, normalized by
+                                                   !! the full horizontal area of a velocity cell [Z ~> m]
+  real,                        intent(in)  :: Dp   !< The larger of the two depths at the edge
+                                                   !! of a velocity cell [Z ~> m]
+  real,                        intent(in)  :: Dm   !< The smaller of the two depths at the edge
+                                                   !! of a velocity cell [Z ~> m]
+  real, dimension(SZK_(GV)+1), intent(out) :: L    !< The fraction of the full cell width that is open at
+                                                   !! the depth of each interface [nondim]
+
+  ! Local variables
+  real :: slope     ! The absolute value of the bottom depth slope across a cell times the cell width [Z ~> m].
+  real :: I_slope   ! The inverse of the normalized slope [Z-1 ~> m-1]
+  real :: Vol_open  ! The cell volume above which it is open [Z ~> m].
+  integer :: K, nz
+
+  nz = GV%ke
+
+  slope = abs(Dp - Dm)
```

Comment:
With this absolute value, this particular subroutine would work if Dp and Dm happened to be switched.  The other subroutines would not work with this reversal, so they had error handling (which has since been removed or commented out).

Reply:
Sorry, I missed the loop here.